### PR TITLE
Improve EventLogTarget installation logic

### PR DESCRIPTION
**Pull request by Coriolanuss**
Remove MaximumKilobytes setting from GetEventLog(), because when app is run without elevated privileges, the method spams exceptions.
Move  MaximumKilobytes setting to installation (and initialization) phase, when the application is typically run with elevated privileges.
Add logic to not override a bigger value of MaximumKilobytes, because the Event log can be used by multiple Sources, each having its own space requirements.

### CODE REVIEW

**Comment by Coriolanuss** [comment]
Hi, can anyone (@304NotModified maybe?) please assist with AppVeyor build? The failed unit test succeeds locally.
The second question: I branched from and merging to the [master branch](https://github.com/NLog/NLog/tree/master) as proposed [here](https://github.com/NLog/NLog#contributing), should it be [dev branch](https://github.com/NLog/NLog)?

**Comment by 304NotModified** [comment]
> The second question: I branched from and merging to the [master branch](https://github.com/NLog/NLog/tree/master) as proposed [here](https://github.com/NLog/NLog#contributing), should it be [dev branch](https://github.com/NLog/NLog)?

It should be `dev` indeed, updated the readme :)

**Comment by 304NotModified** [comment]
> The failed unit test succeeds locally.

This is an unstable (integration) test. That has been fixed in "dev" :)

File: src/NLog/Targets/EventLogTarget.cs [hunk 1]
```diff
@@ -194,10 +194,8 @@ public int MaxMessageLength
         /// <param name="installationContext">The installation context.</param>
         public void Install(InstallationContext installationContext)
         {
-            var fixedSource = GetFixedSource();
-
-            //always throw error to keep backwardscomp behavior.
-            CreateEventSourceIfNeeded(fixedSource, true);
+            // always throw error to keep backwards compatible behavior.
+            CreateEventSourceIfNeeded(GetFixedSource(), true);
```
Comment (304NotModified) [inline src/NLog/Targets/EventLogTarget.cs]: Why is this changed?

File: src/NLog/Targets/EventLogTarget.cs [hunk 2]
```diff
@@ -420,22 +401,18 @@ private void CreateEventSourceIfNeeded(string fixedSource, bool alwaysThrowError
                     {
                         // re-create the association between Log and Source
                         EventLog.DeleteEventSource(fixedSource, MachineName);
-                        var eventSourceCreationData = new EventSourceCreationData(fixedSource, Log)
-                        {
-                            MachineName = MachineName
-                        };
 
-                        EventLog.CreateEventSource(eventSourceCreationData);
+                        EventLog.CreateEventSource(new EventSourceCreationData(fixedSource, Log) { MachineName = MachineName });
```
Comment (304NotModified) [inline src/NLog/Targets/EventLogTarget.cs]: Why is this changed? I find it less readable now.

File: src/NLog/Targets/EventLogTarget.cs [hunk 3]
```diff
@@ -244,20 +242,7 @@ protected override void InitializeTarget()
         {
             base.InitializeTarget();
 
-            var fixedSource = GetFixedSource();
-
-            if (string.IsNullOrEmpty(fixedSource))
-            {
-                InternalLogger.Debug("EventLogTarget(Name={0}): Skipping creation of event source because it contains layout renderers", Name);
```
Comment (304NotModified) [inline src/NLog/Targets/EventLogTarget.cs]: why is this warning being removed?

File: src/NLog/Targets/EventLogTarget.cs [hunk 4]
```diff
@@ -244,20 +242,7 @@ protected override void InitializeTarget()
...
-            }
-            else
-            {
-                var currentSourceName = EventLog.LogNameFromSourceName(fixedSource, MachineName);
```
Comment (304NotModified) [inline src/NLog/Targets/EventLogTarget.cs]: not sure if we need to remove this.

File: src/NLog/Targets/EventLogTarget.cs (hunk 1 above)
Comment (Coriolanuss) [reply src/NLog/Targets/EventLogTarget.cs]: 1) Improves readability: "backwardscomp" → "backwards compatible".
2) Inlined temporary variable to simplify the method.

File: src/NLog/Targets/EventLogTarget.cs (hunk 2 above)
Comment (Coriolanuss) [reply src/NLog/Targets/EventLogTarget.cs]: Guess it's a matter of taste, I don't insist on this change. Same for ln:438.

File: src/NLog/Targets/EventLogTarget.cs (hunk 3 above)
Comment (Coriolanuss) [reply src/NLog/Targets/EventLogTarget.cs]: Removes duplicate code. The exact same check is performed farther in `CreateEventSourceIfNeeded()`.

File: src/NLog/Targets/EventLogTarget.cs (hunk 4 above)
Comment (Coriolanuss) [reply src/NLog/Targets/EventLogTarget.cs]: The same as for the check on the ln:249. Also, additional checks should not be expected for the method with the suffix `IfNeeded`.

**Comment by Coriolanuss** [comment]
> > The second question: I branched from and merging to the [master branch](https://github.com/NLog/NLog/tree/master) as proposed [here](https://github.com/NLog/NLog#contributing), should it be [dev branch](https://github.com/NLog/NLog)?
> 
> It should be `dev` indeed, updated the readme :)

Is another pull request to _dev_ branch needed, or it can be done in this one?

**Comment by Coriolanuss** [comment]
Created another pull request #3017 to merge into _dev_.
